app/src/main/java/hacking/to/the/gate/Jet.java

```diff
@@ -13,10 +13,32 @@
  */
 public class Jet implements Hittable {
 
+    private Effect effect;
     public CircleCollider getCollider(){
         return collider;
     }
 
+    public void onCollision(Hittable h){
+        float curHealth = getHealth();
+        if(h instanceof PowerUp){
+            curHealth += PowerUp.POWERUP_HEAL;
+        }
+        else if(h instanceof Bullet){
+            curHealth -= ((Bullet) h).getDamage();
+        }
+        setHealth(curHealth);
+        if(curHealth < 0) {
+            setDead(true);
+            if(!mIsPlayer) {
+                effect.effectOfKillingEnemy((int) getSelfPosition().getPositionX());
+            }
+        }
+    }
+
+    public void register(Effect e){
+        effect = e;
+
```

Comment:
Should use a more specific name for this method.
Following the Android naming convention, it might be good to call it as setJetLifeCycleListener().